src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,101 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_ELEMENTS = 10;
+    private T[] array;
+    private int size;
+
+    public ArrayList() {
+        array = (T[]) new Object[DEFAULT_ELEMENTS];
+        size = 0;
+    }
 
     @Override
     public void add(T value) {
-
+        if (array.length == size) {
+            grow();
+        }
+        array[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayIndexOutOfBoundsException("Incorrect index");
+        }
+        if (array.length == size) {
+            grow();
+        }
+        System.arraycopy(array, index, array, index + 1, size - index);
+        array[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        addition(list);
     }
 
     @Override
     public T get(int index) {
-        return null;
+        indexCheck(index);
+        return array[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        indexCheck(index);
+        array[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        indexCheck(index);
+        T returnedValue = array[index];
+        System.arraycopy(array, index + 1, array, index, size - index - 1);
+        size--;
+        return returnedValue;
     }
 
     @Override
     public T remove(T t) {
-        return null;
+        for (int i = 0; i < size; i++) {
+            if (array[i] == null || array[i].equals(t)) {
```

Comment:
this is not 100% right

please remember how we compared objects in the previous tasks (for example, keys in storage)